rhtconsulting/rhc-ose
# Securing the registry ansible role.

#### What does this PR do?

Ansible role for securing OSE internal docker registry. This role will secure the registry with a provided certificate else this role will generate a certificate to use for securing.
#### How should this be manually tested?

On an environment with a running docker registry already deployed. (tested on 3.1, however 3.2 shouldn't change anything) fill out the variables in the `secure-registry.yaml` to your specifications for securingthe registry. Then run the playbook:
`ansible-playbook -i <inventory> rhc-ose/rhc-ose-ansible/roles/secure-registry/test/secure-registry.yaml`

Note: You may need to issue a redeploy of the registry. The ansible role restarts the docker service after configuring certificates, which may break current deployments kicked off by config changes. 

To validate:

`oc logs <docker registry pod>` and ensure its running in TLS mode:
`level=info msg="listening on :5000, tls"`

Authenticate to the docker registry and do a pull-tag-push.
`$ docker pull busybox
$ docker tag docker.io/busybox 172.30.124.220:5000/<project>/busybox
$ docker push 172.30.124.220:5000/openshift/busybox`
#### Is there a relevant Issue open for this?

N/A
#### Who would you like to review this?

/cc @etsauer  @sabre1041 @oybed
